manifests/config/systemd.pp

```diff
@@ -0,0 +1,69 @@
+# == Class: marathon::config::systemd
+#
+class marathon::config::systemd{
+
+  $conf_dir = $marathon::config::conf_dir
+  $owner = $marathon::config::owner
+  $group = $marathon::config::group
+  $master = $marathon::config::master
+  $zookeeper = $marathon::config::zookeeper
+  $options = $marathon::config::options
+  $ulimit = $marathon::config::ulimit
+  $env_var = $marathon::config::env_var
+  $java_home = $marathon::config::java_home
+  $java_opts = $marathon::config::java_opts
+  $java_extra_opts = $marathon::config::java_extra_opts
+
+  $ensure_service_file = $ulimit ? {
+    undef => 'absent',
+    default => 'present'
+  }
+
+  if ($master) {
+    $master_options = { 'master' => $master }
+  } else {
+    $master_options = { }
+  }
+  if ($zookeeper) {
+    $zk_options = { 'zk' => $zookeeper }
+  } else {
+    $zk_options = { }
+  }
+
+  $all_options = merge($options, $master_options, $zk_options)
+
+  # wipe old config dir, everything is configured in /etc/sysconfig/marathon
+  file { $conf_dir:
+    ensure  => 'absent',
+    recurse => true,
+    purge   => true,
+    force   => true,
+  }
+
+  # wipe old config file, everything is configured in /etc/sysconfig/marathon
+  file { '/etc/default/marathon':
+    ensure => 'absent',
+  }
+
+  file { '/etc/sysconfig/marathon':
+    ensure  => 'present',
+    content => template('marathon/sysconfig.erb'),
+    owner   => 'root',
+    group   => 'root',
+    mode    => '0644',
+  }
+
+  file { '/etc/systemd/system/marathon.service':
+    ensure  => $ensure_service_file,
+    content => template('marathon/marathon.service.erb'),
+    owner   => 'root',
+    group   => 'root',
+    mode    => '0644',
+    notify  => Exec['systemctl-daemon-reload_marathon'],
+  }
+
+  exec { 'systemctl-daemon-reload_marathon':
```

Comment:
This seems like a thing that Puppet should handle? Or is there some standard module for handling systemd? I don't like the idea of writing `exec` commands to refresh services.